File: src/Models/db/lyda/Comment.ts
import {User} from "./User";
import {Track} from "./Track";
import {InteractionMetadata} from "../../InteractionMetadata";
import {CommentLike} from "./CommentLike";
import {CommentReport} from "./CommentReport";

export interface Comment {
    comments?: Comment[];
    user?: User;
    track?: Track;
    canEdit?: boolean;
    likes?: InteractionMetadata<CommentLike>;
    reports?: CommentReport[];
    report_count?: number;
    reported_by_me?: boolean;
    id: number;
    parent_id: number;
    track_id: number;
    user_id: number;
    content: string;
    created_at: Date;
    potentially_harmful: boolean;
    hidden: boolean;
}

File: kotlin/com/example/lydaandroid/model/Comment.kt
package com.example.lydaandroid.model

import java.util.Date

data class Comment(
    val id: Int,
    val parent_id: Int? = null,
    val track_id: Int,
    val user_id: Int,
    val content: String,
    val created_at: String,
    val potentially_harmful: Boolean,
    val hidden: Boolean,
    val comments: List<Comment>? = null,
    val user: User? = null,
    val track: Track? = null,
    val canEdit: Boolean? = null
)
